sections/Philosophy.tex

```diff
@@ -0,0 +1,60 @@
+From \myref{sec:state_of_the_art} we know that several libraries and languages allows the programmer to explicitly designate workloads to the GPU however this often requires explicit memory handling as well.
+GAMBLE is meant to be used for computations which has a need for more computational power; due to this the language should not be too unfamiliar.
+Keeping GAMBLE familiar to other languages reduces the time required to familiarize one with GAMBLE.
+This is done by using a C-like syntax, while stripping GAMBLE of things deemed not necessary for computational needs, as will be documented in the rest of this chapter.
+As the GPU is the resource being used to achieve more computing power, the data computed must also be applicable to the niche of the GPU, i.e. the data must be parallelisable.
+This basic need will influence how data is represented, and also puts focus onto matrices and vector calculations, which often can be paralellised.
+As it follows from this philosophy, some criteria for GAMBLE exists.
+\begin{itemize}
+	\item Allow the programmer to use the GPU without it being inconvenient.
+	\item Avoid implementing unnecessary data types and features.
+	\item Let the language be somewhat familiar to read and use.
+	\item Let the language promote read- and write-ability.
+\end{itemize}
+
+\textbf{Allow the programmer to use the GPU without it being inconvenient}
+
+Due to GAMBLE being focused on numerical computations, allowing the programmer to focus on managing the mathematical aspects is key.
+Therefore having the programmer control the data flow seems an unnecessary distraction.
+GAMBLE takes care of designating the computations to the right processor, whether it be the GPU or the CPU, as such any inconvenience in that process is removed from the programmer whose focus can be solely on the mathematics.
+
+\textbf{Avoid implementing unnecessary data types and features}
+
+When deciding what any language should offer its user, it is important to keep in mind what is the purpose of the language.
+As GAMBLEs purpose is to use the GPU to for calculations which can be parallelised implementing features or data types that do not hold any regard to this aspect would clutter the language.
+Additionally GAMBLE should not try to adapt itself towards purposes for which it is not designed, and example of excluding such features is the fact that strings are not part of the language, this choice and others like it are further documented in this chapter.
+
+\textbf{Let the language be somewhat familiar to read and use}
+
+As mentioned the main purpose of GAMBLE is to use the GPU for computations, and is focused on doing computations, not developing new software.
+As such GAMBLE would most often be used where this niche is required.
+It may even be likely that it is not used when developing and algorithm to do a thing, but first used once the algorithm is complete, and can be applied to bigger sets of data.
```

Comment:
developing AN algorithm to do a thing.
Ret til an, og hvad betyder a thing, præcision! :dancer: